components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/tokenprocessor/TokenValidationProcessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.tokenprocessor;
+
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+
+/**
+ * Token validation Processor.
+ */
+public interface TokenValidationProcessor {
+
+    /**
+     * Validate token.
+     *
+     * @param accessToken    access token
+     * @param includeExpired include expired tokens
+     * @return AccessTokenDO
+     * @throws IdentityOAuth2Exception if an error occurred while validating the token
+     */
+    AccessTokenDO validateToken(String accessToken, boolean includeExpired) throws IdentityOAuth2Exception;
```

Comment:
In almost all the places we have replaced findAccessToken with validateToken. In these places, we are only retrieving the token and there is no validation happening. Can we rename this method (and also the class) to better reflect what we are doing?

Currently it doesn't seem intuitive that we need to call a validate method in order to retrieve a token

Reply:
Hi @janakamarasena 

Eventhough the name of the previous method was `findAccessToken`, underlying it was returning a AccessTokenDAO if and only if the token is in active or expired state in the database. 
If its a revoked JTI/opaque, the return value will be null. So in our in memory implementation of the method (when token persistence is removed) , we are doing the same by validating the token sig, expiry time, consumer key and  checking if the JTI is already revoked through direct or indirect revocation events.

Agree with your point on confusion when compared to the old method name as it does not reflect the fact that that it deals with returning a token. Hence how about following naming.

```
public interface AccessTokenProvider {
    AccessTokenDO getVerifiedAccessToken(String tokenIdentifier);
}
```

Reply:
+1 for the suggested names